File: openapi-generator/clojure/src/samsara_api/specs/vehicle_harsh_event_response_location.clj
(ns samsara-api.specs.vehicle-harsh-event-response-location
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            )
  (:import (java.io File)))


(def vehicle-harsh-event-response-location-data
  {
   (ds/opt :address) string?
   (ds/opt :latitude) string?
   (ds/opt :longitude) string?
   })

(def vehicle-harsh-event-response-location-spec
  (ds/spec
    {:name ::vehicle-harsh-event-response-location
     :spec vehicle-harsh-event-response-location-data}))

File: openapi-generator/php-ze-ph/src/App/DTO/VehicleHarshEventResponseLocation.php
<?php
declare(strict_types=1);

namespace App\DTO;

use Articus\DataTransfer\Annotation as DTA;

/**
 */
class VehicleHarshEventResponseLocation
{
    /**
     * Address of location where the harsh event occurred
     * @DTA\Data(field="address", nullable=true)
     * @DTA\Validator(name="Type", options={"type":"string"})
     * @var string
     */
    public $address;
    /**
     * Latitude of location where the harsh event occurred
     * @DTA\Data(field="latitude", nullable=true)
     * @DTA\Validator(name="Type", options={"type":"string"})
     * @var string
     */
    public $latitude;
    /**
     * Longitude of location where the harsh event occurred
     * @DTA\Data(field="longitude", nullable=true)
     * @DTA\Validator(name="Type", options={"type":"string"})
     * @var string
     */
    public $longitude;
}
